### ISSUE TITLE
Problem running scenario 2 on Ubuntu with snap-based Node.js

### ISSUE THREAD

**Issue by DavidBruant** [Agoric/cosmic-swingset#71]
When trying `make scenario2-setup`, i get:

```
rm -r ~/.ag-chain-cosmos
rm: impossible de supprimer '/home/david/.ag-chain-cosmos': Aucun fichier ou dossier de ce type
make: [Makefile:42: scenario2-setup] Error 1 (ignorée)
rm ag-cosmos-chain-state.json
rm: impossible de supprimer 'ag-cosmos-chain-state.json': Aucun fichier ou dossier de ce type
make: [Makefile:43: scenario2-setup] Error 1 (ignorée)
./lib/ag-chain-cosmos init --chain-id=agoric
Error: /snap/core/current/usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.22' not found (required by /home/david/projects/cosmic-swingset/build/Release/agcosmosdaemon.node)
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:977:18)
make: *** [Makefile:44: scenario2-setup] Error 1
```

The problems with the `rm`s are expected i imagine. However, it's less clear to me the other error (about `GLIBCXX_3.4.22`) is expected

**Comment by DavidBruant** [comment]
> It indicates that the libstdc++ library you built the node addon (agcosmosdaemon.node) against is not the same one you're running against.

ok, that makes sense to me

> I'd recommend rerunning npm run build at the top of cosmic-swingset in the same environment you're using to run the Makefile. That should rebuild agcosmosdaemon.node and fix your problem.

I tried and it didn't solve the problem
The presence of `/snap` on the path suggests to me that in one command it's using the [snap](https://snapcraft.io/) version and in another command it uses another non-snap version available somewhere on my computer
(i have an equivalent problem with npm [run scripts](https://docs.npmjs.com/cli/run-script) which use a different version of node than the one i installed via snap)
I do not know what to do to solve this problem yet, but it gives me a place to look at

> set your LANG=C environment variable before you run the command whose output you're pasting

Oh ok, i wasn't aware that was a possibility. I'll do that from now on, thanks!

**Comment by DavidBruant** [comment]
> It looks like the npm executable doesn't run the snap version of Node.

That part i know (and it only emits a warning which is always irrelevant in the cases i care about). 

What i was trying to say is that i've noticed problems of things being installed and other things saying "not found" and the underlying matter was a snap/non-snap thing
and i believe this is the problem i'm having now with `make scenario2-setup`

I have the feeling that `npm run build` used the snap version of something and `make scenario2-setup` does not (or the other way around)

**Comment by DavidBruant** [comment]
ok, this is way too hard for me. It'd be probably days for me to figure this one out on my own
But i'm stuck on this scenario and i imagine i'll have the same problem with scenario0

I'm pondering hiding all the Golang+Python+(Node C++ addon) work (and probably a good part of the Makefile) in a dockerfile

**Comment by DavidBruant** [comment]
I forgot to mention that before this error, `make scenario2-setup` lead to an [error i had already had in the past](https://github.com/Agoric/cosmic-swingset/issues/51#issuecomment-519971480), so i had to run `npm run build` in order to move forward

I don't know yet how to do that, but i think a solution could be that `agcosmosdaemon.node` could be built inside a docker container (with the proper `/usr/lib/x86_64-linux-gnu/libstdc++.so.6`), then run inside a docker container (with the same `/usr/lib/x86_64-linux-gnu/libstdc++.so.6`)

**Comment by DavidBruant** [comment]
I reinstalled a non-snap version of Node.js
I'm having other problems, but i think this issue can be closed (i created Agoric/cosmic-swingset#75 with that intention)

**Comment by warner** [comment]
in the old repo. this was cosmic-swingset issue 71
